### PR TITLE
add Gutenberg block relations processing

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: inc/Smartling/Helpers/RelativeLinkedAttachmentCoreHelper.php [hunk 1]
```diff
@@ -100,21 +91,13 @@ protected function processString(&$stringValue): void
...
-        if (0 < preg_match_all(self::CORE_GUTENBERG_BLOCK, $stringValue, $matches)) {
-            foreach ($matches[1] as $key => $_) {
-                $info = $this->processGutenbergBlock($matches[1][$key], $matches[2][$key]);
-                if ($info !== null) {
-                    $stringValue = str_replace("<!-- wp:core/{$matches[1][$key]} {$matches[2][$key]} -->", "<!-- wp:{$matches[1][$key]} {$info->getReplacement()} -->", $stringValue);
-                    $replacer->addReplacementPair(new ReplacementPair("wp-image-{$info->getSourceId()}", "wp-image-{$info->getTargetId()}"));
-                }
-            }
-        }
-
```
Comment (vsolovei-smartling) [inline inc/Smartling/Helpers/RelativeLinkedAttachmentCoreHelper.php:117]: This was moved to GutenbergBlockHelper, and further upgraded

File: inc/Smartling/Replacers/ContentIdReplacer.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                    $this->contentType,
+                    $sourceBlogId,
+                    (int)$value,
+                    $targetBlogId,
+                    $submission->getJobInfoWithBatchUid()
+                )->getTargetId();
+
+                if ($dataType === 'string') {
+                    $value = (string)$targetId;
+                }
```
Comment (PavelLoparev) [inline inc/Smartling/Replacers/ContentIdReplacer.php:86]: I don't get why in some cases we need to cast value to string

File: inc/Smartling/WP/View/AdminPage.php [hunk 3]
```diff
@@ -53,7 +53,7 @@
         <?php $filterTable->display(); ?>
     </form>
 
-    <h3>Custom media attachment rules</h3>
+    <h3>Gutenberg block rules</h3>
```
Comment (PavelLoparev) [inline inc/Smartling/WP/View/AdminPage.php:56]: So we dropped media attachments support or now we have some generic approach for all attachments?

**Review by PavelLoparev — COMMENTED** [review]
I think demo is needed. Additionally, seems like tests are missing

**Review by PavelLoparev — COMMENTED** [review]
I think demo is needed. Additionally, seems like tests are missing

**Pull request merged**